app/main.py

```diff
@@ -1,2 +1,49 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.dict_len = 8
+        self.length = 0
+        self.element = [None] * self.dict_len
+
+    def __setitem__(self, key, value):
+        if len(self) == int(self.dict_len * 2 / 3):
+            self.resize()
+
+        hash_key = hash(key)
+        new_el = (hash_key, key, value)  # store: hash, key, value
+        i = hash_key % self.dict_len
+
+        while True:
+            if self.element[i] is None:
+                self.element[i] = new_el
+                self.length += 1
+                break
+            elif hash_key == self.element[i][0] and key == self.element[i][1]:
+                self.element[i] = new_el
+                break
+            else:
+                i = (i + 1) % self.dict_len
+
+    def __getitem__(self, key):
+        hash_key = hash(key)
+        i = hash_key % self.dict_len
+
+        if self.element[i] is None:
+            raise KeyError
+
+        while True:
```

Comment:
What if the key isn't found? 
It will lead to the infinite loop :) 
If you visit an empty cell you can raise a KeyError exception